pkg/workflow/def/workflow/fork_join_dynamic.go

```diff
@@ -0,0 +1,56 @@
+package workflow
+
+import "github.com/conductor-sdk/conductor-go/pkg/http_model"
+
+type DynamicForkTask struct {
+	Task
+	preForkTask TaskInterface
+	join        JoinTask
+}
+
+func DynamicFork(taskRefName string, forkPrepareTask TaskInterface) *DynamicForkTask {
+	return &DynamicForkTask{
+		Task: Task{
+			name:              taskRefName,
+			taskReferenceName: taskRefName,
+			description:       "",
+			taskType:          FORK_JOIN_DYNAMIC,
+			optional:          false,
+			inputParameters:   map[string]interface{}{},
+		},
+		preForkTask: forkPrepareTask,
+	}
+}
+
+func DynamicForkWithJoin(taskRefName string, forkPrepareTask TaskInterface, join JoinTask) *DynamicForkTask {
+	return &DynamicForkTask{
+		Task: Task{
+			name:              taskRefName,
+			taskReferenceName: taskRefName,
+			description:       "",
+			taskType:          FORK_JOIN_DYNAMIC,
+			optional:          false,
+			inputParameters:   map[string]interface{}{},
+		},
+		preForkTask: forkPrepareTask,
+		join:        join,
+	}
+}
+
+func (task *DynamicForkTask) toWorkflowTask() []http_model.WorkflowTask {
+
+	forkWorkflowTask := task.Task.toWorkflowTask()[0]
```

Comment:
check for length?

Reply:
toWorkflowTask in the Task always returns an array with 1 element, so no need.